pkg/proc/target_exec.go

```diff
@@ -1014,8 +1018,8 @@ func stepOutReverse(p *Target, topframe, retframe Stackframe, sameGCond ast.Expr
 	return err
 }
 
-// onNextGoroutine returns true if this thread is on the goroutine requested by the current 'next' command
-func onNextGoroutine(thread Thread, breakpoints *BreakpointMap) (bool, error) {
+// OnNextGoroutine returns true if this thread is on the goroutine requested by the current 'next' command
+func OnNextGoroutine(thread Thread, breakpoints *BreakpointMap) (bool, error) {
```

Comment:
I'm not fond of exposing this function. Maybe KeepsSteppingBreakpoints should be more granular than just a boolean.

Reply:
Good point.

The two cases for DAP that we want to keep the stepping breakpoint are:
1. halt request
2. log points (which can be marked as tracepoints)

I can change this to be a setting that is more granular to cover these two cases.

Though I wonder if tracepoints should actually never clear stepping breakpoints?

Reply:
So:

```
type KeepSteepingBreakpoints uint8
const` (
    HaltKeepsSteppingBreakpoints KeepSteppingBreakpoints = 1 << iota
    TracepointKeepsSteppingBreakpoints
    // others?
)
```

> Though I wonder if tracepoints should actually never clear stepping breakpoints?

I think we could do that. The argument for keeping it this way is that it's been like this for a long time, but I doubt it would hurt anything.

Reply:
I switched it to preserver the old behavior for tracepoints.  and updated the configuration to match @aarzilli comment above.